File: 0769.最多能完成排序的块/0769-最多能完成排序的块.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        ret = 0
        until_index = 0
        for index,num in enumerate(arr):
            until_index = max(until_index, arr[index])
            if index == until_index:
                ret += 1
        return ret
            

File: 0769.最多能完成排序的块/0769-最多能完成排序的块.rs
impl Solution {
    pub fn max_chunks_to_sorted(arr: Vec<i32>) -> i32 {
        let mut ret = 0;
        let mut until_index = 0usize;
        for (index, num) in arr.iter().enumerate() {
            until_index = until_index.max(*num as usize);
            if index == until_index {
                ret += 1;
            }
        }
        
        
        ret
    }
}